### PR TITLE
fixed change detection

### PR DESCRIPTION
**Pull request by chensation**
Fixed the delayed change detection bug present in PowerShell commands.  See bug within the attached image.
new commands

### CODE REVIEW

File: src/SfxWeb/src/app/modules/powershell-commands/command/command.component.ts [hunk 1]
```diff
@@ -42,12 +43,20 @@ export class CommandComponent implements OnInit{
...
     for (let name in this.requiredInputs?.controls) {
-      if (!this.requiredInputs.controls[name]?.valid)
+      this.command.getParam(name).value = this.requiredInputs.controls[name].value;
+      if (!this.requiredInputs.controls[name]?.valid){
         invalids.push(name);
+      }
     }
     this.invalidInputs = invalids.join(', ');
+
+    for (let name in this.optionalInputs?.controls) {
```
Comment (jeffj6123) [inline src/SfxWeb/src/app/modules/powershell-commands/command/command.component.ts:56]: when is optionalinputs null?

File: src/SfxWeb/src/app/modules/powershell-commands/command/command.component.ts [hunk 2]
```diff
@@ -42,12 +43,20 @@ export class CommandComponent implements OnInit{
     return this.inputForm.controls['optionalInputs'] as UntypedFormGroup;
   }
 
-  updateInvalidInputText() {
+  updateInputs() {
     let invalids = [];
     for (let name in this.requiredInputs?.controls) {
```
Comment (jeffj6123) [inline src/SfxWeb/src/app/modules/powershell-commands/command/command.component.ts:48]: when is this null?

**Review by jeffj6123 — CHANGES_REQUESTED** [review]

File: src/SfxWeb/src/app/modules/powershell-commands/command/command.component.ts (hunk 1 above)
Comment (chensation) [reply src/SfxWeb/src/app/modules/powershell-commands/command/command.component.ts]: they were erroring out on page load as inputForm was not set yet, undefined/null was being returned by the getters. Checking again today that doesn't seem to be the case anymore(???). Removed.

File: src/SfxWeb/src/app/modules/powershell-commands/command/command.component.ts (hunk 2 above)
Comment (chensation) [reply src/SfxWeb/src/app/modules/powershell-commands/command/command.component.ts]: Removed.

**Pull request merged**